File: test_cgu.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import TimeoutException

def run_test_cgu():
    driver = webdriver.Chrome()
    driver.maximize_window()

    try:
        # Charger la page principale
        driver.get("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com")
        print("Test du lien 'CGU'...")

        try:
            # Localiser le lien 'CGU' et cliquer dessus
            cgu_link = WebDriverWait(driver, 10).until(
                EC.element_to_be_clickable((By.LINK_TEXT, "CGU"))
            )
            cgu_link.click()

            # Attendre que l'URL change pour correspondre à celle des CGU
            WebDriverWait(driver, 10).until(
                EC.url_contains("terms-and-conditions")
            )
            print("✅ Lien 'CGU' cliqué, redirection réussie.")

            # Retourner à la page principale
            driver.back()
            WebDriverWait(driver, 10).until(
                EC.url_contains("init")
            )
            print("✅ Retour à la page principale réussi.")
        except TimeoutException:
            print("❌ Test du lien 'CGU' échoué : la redirection ou le retour n'a pas fonctionné.")
    except Exception as e:
        print(f"Erreur inattendue : {e}")
    finally:
        driver.quit()

# Appeler la fonction pour exécuter le test
if __name__ == "__main__":
    run_test_cgu()

File: test_cgu.ts
import { chromium } from 'playwright';

export async function runTestCGU() {
    const browser = await chromium.launch({ headless: false });
    const context = await browser.newContext();
    const page = await context.newPage();

    try {
        // Charger la page principale
        await page.goto("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com");
        console.log("Test du lien 'CGU'...");
        await page.waitForTimeout(2000);
        // Localiser le lien 'CGU' et cliquer dessus
        const cguLink = await page.waitForSelector("text=CGU", { timeout: 10000 });
        await cguLink.click();
        console.log("✅ Lien 'CGU' cliqué.");
        await page.waitForTimeout(2000);
        // Attendre que l'URL contienne "terms-and-conditions"
        await page.waitForURL(/terms-and-conditions/, { timeout: 10000 });
        console.log("✅ Redirection vers 'CGU' réussie.");
        await page.waitForTimeout(2000);
        // Vérifier que la page chargée contient bien le texte attendu
        await page.waitForSelector("h1", { timeout: 10000 });  // Modifier si nécessaire
        console.log("✅ Contenu de la page CGU détecté.");
        // Retourner à la page principale
        await page.goBack();
        await page.waitForURL(/init/, { timeout: 10000 });
        console.log("✅ Retour à la page principale réussi.");
        await page.waitForTimeout(2000);
    } catch (error) {
        console.error("❌ Test du lien 'CGU' échoué :", error);
    } finally {
        await browser.close();
    }
}

// if (require.main === module) {
//     runTestCGU();
// }
